src/sharding/ShardingManager.js

```diff
@@ -262,9 +266,12 @@ class ShardingManager extends EventEmitter {
       return Promise.reject(new Error('SHARDING_SHARD_NOT_FOUND', shard));
     }
 
-    const promises = [];
-    for (const sh of this.shards.values()) promises.push(sh[method](...args));
-    return Promise.all(promises);
+    return Promise.all(
+      this.shards.reduce((promises, _shard) => {
+        promises.push(_shard[method](...args));
+        return promises;
+      }),
+    );
```

Comment:
I knew I forgot to mention one yesterday. But here it is. This line is also missing an initial value.